### PR TITLE
PR for issue #398.

### PR DESCRIPTION
**Pull request by markusgumbel**
Fixed issue #398. 
 * Changed file `char-rnn.jl` according to description/sugesstion in issue text.
 * Set Julia version to 1.10. The Julia 1.10 runtime was not able to run the 1.6 version.

### CODE REVIEW

File: text/char-rnn/char-rnn.jl [hunk 1]
```diff
@@ -66,8 +66,10 @@ function getdata(args::Args)
     
     ## Partitioning the data as sequence of batches, which are then collected 
     ## as array of batches
-    Xs = partition(batchseq(chunk(text, args.batchsz), stop), args.seqlen)
-    Ys = partition(batchseq(chunk(text[2:end], args.batchsz), stop), args.seqlen)
+    Xtext = [collect(t) for t in chunk(text, args.batchsz)]
+    Ytext = [collect(t) for t in chunk(text[2:end], args.batchsz)]
+    Xs = partition(batchseq(Xtext, stop), args.seqlen)
+    Ys = partition(batchseq(Ytext, stop), args.seqlen)
```
Comment (ToucheSir) [inline text/char-rnn/char-rnn.jl:72]: Can you explain the reasoning behind this change? Generally, I would not expect `collect` to be required here.

Comment (markusgumbel) [reply text/char-rnn/char-rnn.jl:72]: The problem seems to be that `chunk(text, args.batchsz)` creates an array of `SubString`s. `batchseq` calls `ndims` on this array. However, `ndims` does not seem to be defined for `SubString`s. The Strings are converted into an array of characters (for which `ndims` is available).
Without the change you get:
ERROR: LoadError: MethodError: no method matching ndims(::SubString{String})

Comment (ToucheSir) [reply text/char-rnn/char-rnn.jl:72]: I see. We should fix `batchseq` to work with strings, but that can happen separately. Thanks for elaborating.

**Review by ToucheSir — APPROVED** [review]
Thanks for updating this!

**Comment by markusgumbel** [comment]
Thank you, too.

**Pull request merged**